src/lib/validation/generic.ts

```diff
@@ -36,3 +37,29 @@ export function validateEmailString (emailString: string): boolean {
         return false;
     }
 }
+
+export function validateCompanyNumberSearchString (str: string): boolean {
+    const searchSchema = z.string()
+        .trim().regex(constants.COMPANY_NUMBER_SEARCH_VALIDATION_REGEX);
+    try {
+        searchSchema.parse(str);
+        return true;
+    } catch (e) {
+        return false;
+    }
+}
+
+export function validatePageNumber (pageNum: number, arrayLength: number, perPage: number): boolean {
+    let maxNumOfPages = 1;
```

Comment:
again you dont need this

Reply:
as above

Reply:
Actually, after checking the details, this is not about what comes from the backend. This validation is to prevent errors from manually changing the page URL in the browser.

Reply:
Saying that, the code is refactored